nexus/db-queries/src/db/datastore/disk.rs

```diff
@@ -285,28 +448,70 @@ impl DataStore {
             disk_type_crucible_dsl::disk_type_crucible
                 .on(dsl::id.eq(disk_type_crucible_dsl::disk_id)),
         )
+        .left_join(
+            disk_type_local_storage_dsl::disk_type_local_storage
+                .on(dsl::id.eq(disk_type_local_storage_dsl::disk_id)),
+        )
         .filter(dsl::time_deleted.is_null())
         .filter(dsl::attach_instance_id.eq(authz_instance.id()))
         .select((
             model::Disk::as_select(),
             Option::<DiskTypeCrucible>::as_select(),
+            Option::<DiskTypeLocalStorage>::as_select(),
         ))
-        .get_results_async(&*self.pool_connection_authorized(opctx).await?)
+        .get_results_async(&*conn)
         .await
         .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
 
         let mut list = Vec::with_capacity(results.len());
 
         for result in results {
             match result {
-                (disk, Some(disk_type_crucible)) => {
+                (disk, Some(disk_type_crucible), None) => {
                     list.push(Disk::Crucible(CrucibleDisk {
                         disk,
                         disk_type_crucible,
                     }));
                 }
 
-                (disk, None) => {
+                (disk, None, Some(disk_type_local_storage)) => {
+                    let local_storage_dataset_allocation = if let Some(
+                        allocation_id,
+                    ) =
+                        disk_type_local_storage
+                            .local_storage_dataset_allocation_id()
+                    {
+                        use nexus_db_schema::schema::local_storage_dataset_allocation::dsl;
+
+                        Some(
+                            dsl::local_storage_dataset_allocation
+                                .filter(
+                                    dsl::id.eq(to_db_typed_uuid(allocation_id)),
+                                )
+                                .select(
+                                    LocalStorageDatasetAllocation::as_select(),
+                                )
+                                .first_async(&*conn)
+                                .await
+                                .map_err(|e| {
+                                    public_error_from_diesel(
+                                        e,
+                                        ErrorHandler::Server,
+                                    )
+                                })?,
+                        )
```

Comment:
style nit, considered obnoxious (take it or leave it): wrapping this whole expression in `Some` feels a bit weird to me, i might prefer to write it with a `let` and then wrap the `let` value in `Some`. that also decreases indentation by one tabstop and might make rustfmt not put the filter/select clauses on new lines:
```suggestion
                        let allocation = dsl::local_storage_dataset_allocation
                            .filter(dsl::id.eq(to_db_typed_uuid(allocation_id)))
                            .select(LocalStorageDatasetAllocation::as_select())
                            .first_async(&*conn)
                            .await
                            .map_err(|e| {
                                public_error_from_diesel(
                                    e,
                                    ErrorHandler::Server,
                                )
                            })?;
                        Some(allocation)
```